app/controllers/restaurants_controller.rb

```diff
@@ -4,7 +4,7 @@ class RestaurantsController < ApplicationController
   # GET /restaurants
   # GET /restaurants.json
   def index
-    render json: Restaurant.all
+    render json: Restaurant.left_joins(:reviews).select("restaurants.*, cast(avg(rating) as integer) as rating").group(:id).order(:name)
```

Comment:
1. this logic should be in the model and not in the controller
2. I don't like the idea of `select *` you van do the avg calculation in the code or create view for that